File: python/templative/lib/create/viewBoxFixer.py
from os import walk
import xml.etree.ElementTree as ET

f = []
path = "C:/Users/User/Documents/git/templative-frontend/python/templative/lib/create/componentTemplates"
for (dirpath, dirnames, filenames) in walk(path):
    
    for filename in filenames:
        filepath = 'C:/Users/User/Documents/git/templative-frontend/python/templative/lib/create/componentTemplates/%s' % filename
        tree = ET.parse(filepath)
        root = tree.getroot()
        
        width = root.get("width", "")
        height = root.get("height", "")
        viewbox = root.get("viewBox", "").split(" ")
        print(width, height, viewbox)
        if width == viewbox[2] and height == viewbox[3]:
            continue
        root.set("viewBox", "0 0 %s %s" %(width, height))
        with open(filepath, "wb") as svgFile:
            svgFile.write(ET.tostring(root))
    
    break

File: src/main/templative/lib/create/viewBoxFixer.js
const fs = require('fs');
const path = require('path');
const { parseStringPromise } = require('xml2js');
const { Builder } = require('xml2js');

async function processFiles() {
  const dirPath = "C:/Users/User/Documents/git/templative-frontend/python/../create/componentTemplates";
  const files = fs.readdirSync(dirPath);

  for (const file of files) {
    const filePath = path.join(dirPath, file);
    const xml = fs.readFileSync(filePath, 'utf-8');
    
    const result = await parseStringPromise(xml);
    const root = result.svg;
    
    const width = root.$.width || "";
    const height = root.$.height || "";
    const viewBox = (root.$.viewBox || "").split(" ");
    console.log(width, height, viewBox);
    
    if (width === viewBox[2] && height === viewBox[3]) {
      continue;
    }
    
    root.$.viewBox = `0 0 ${width} ${height}`;
    
    const builder = new Builder();
    const updatedXml = await builder.buildObject(result);
    
    fs.writeFileSync(filePath, updatedXml);
  }
}

processFiles();